coherence-data/src/main/java/io/micronaut/coherence/data/ops/DefaultCoherenceRepositoryOperations.java

```diff
@@ -239,14 +257,16 @@ protected void setSession(final String sessionName) {
 
     @NonNull
     @Override
+    @PersistEventSource
     public <T> T persist(@NonNull final InsertOperation<T> operation) {
         T entity = operation.getEntity();
-        T result = (T) getNamedMap().put(getId(entity), entity);
-        return result == null ? entity : result;
+        getNamedMap().put(getId(entity), entity);
+        return entity;
     }
 
     @NonNull
     @Override
+    @UpdateEventSource
     public <T> T update(@NonNull final UpdateOperation<T> operation) {
         T entity = operation.getEntity();
         getNamedMap().remove(getId(entity), entity);
```

Comment:
Why are we calling `remove` here? How is that an "update"?

Reply:
Bug.  Will address.